File: my_av.py
#!/usr/bin/python3
"""Main module of homework solving task 1 and task 2"""

import csv
from urls import predict_url, OUTPUT_URLS_PATH, INPUT_URLS_PATH
from traffic import predict_traffic, OUTPUT_TRAFFIC_PATH, INPUT_TRAFFIC_PATH

def check_urls(input_file_path, output_file_path):
    """Solving task 1"""
    # reads data from the input file and opens the output file for writing
    with open(input_file_path, "r", encoding="utf8") as urls, \
        open(output_file_path, "w", encoding="utf8") as output_file:
        for url in urls:
            # removes the trailing newline
            url = url.strip().lower()

            # gets the prediction
            prediction = predict_url(url)

            # writes the result in the output file
            output_file.write(f"{prediction}\n")

def check_traffic(input_file_path, output_file_path):
    """Solving task 2"""
    # reads data from the input file and opens the output file for writing
    with open(input_file_path, "r", encoding="utf8") as traffic, \
        open(output_file_path, "w", encoding="utf8") as output_file:
        csv_reader = csv.reader(traffic, delimiter=',')
        next(csv_reader) # gets rid of the first row of csv file
        for traffic in csv_reader:
            # writes the result in the output file
            output_file.write(f"{predict_traffic(traffic)}\n")


def main():
    """Main function to be called when file runs as script"""
    check_urls(INPUT_URLS_PATH, OUTPUT_URLS_PATH)
    check_traffic(INPUT_TRAFFIC_PATH, OUTPUT_TRAFFIC_PATH)

if __name__ == "__main__":
    main()
    

File: my_av.c
#include <stdio.h>
#include "urls.h"
#include "traffic.h"

// transforms string to lowercase
void string_to_lower(char *string)
{
	int size = strlen(string);
	for (int i = 0; i < size; ++i)
		string[i] = tolower(string[i]);
}

int main(void)
{
	FILE *input_urls = fopen(INPUT_URLS_PATH, "r");
	FILE *output_urls = fopen(OUTPUT_URLS_PATH, "w");

	char input[1024];

	if (input_urls) {
		while (fgets(input, sizeof(input), input_urls)) {
			// remove trailing \n (newline)
			input[strlen(input) - 1] = '\0';

			// make the input lowercase
			string_to_lower(input);

			// print the output based on the prediction
			fprintf(output_urls, "%d\n", predict_url(input));
		}
		// close the input file
		fclose(input_urls);
	}

    // close the output file
	if (output_urls)
		fclose(output_urls);

    // open the files
	FILE *input_traffic = fopen(INPUT_TRAFFIC_PATH, "r");
	FILE *output_traffic = fopen(OUTPUT_TRAFFIC_PATH, "w");

	if (input_traffic) {
		// get rid of the first line of csv file
		fgets(input, sizeof(input), input_traffic);

		// read the rest of the lines
		while (fgets(input, sizeof(input), input_traffic)) {
			// remove trailing \n (newline)
			input[strlen(input) - 1] = '\0';

			// print the output based on the prediction
			fprintf(output_traffic, "%d\n", predict_traffic(input));
		}
		fclose(input_traffic);
	}

    // close the output file
	if (output_traffic)
		fclose(output_traffic);

	return 0;
}
